pkg/apis/core/types.go

```diff
@@ -2887,6 +2887,29 @@ type PodSpec struct {
 	// All topologySpreadConstraints are ANDed.
 	// +optional
 	TopologySpreadConstraints []TopologySpreadConstraint
+	// Specifies the OS on which pod intends to run. This field is needed during the API server admission time to
+	// distinguish among various node OSes in a heterogeneous Kubernetes cluster.
+	// +optional
+	OS *PodOS
+}
+
+// OSName is the set of OS'es that can be used in OS.
+type OSName string
+
+// These are valid values for OSName
+const (
+	Linux   OSName = "linux"
+	Windows OSName = "windows"
+)
+
+// PodOS defines the OS parameters of a pod.
+type PodOS struct {
+	// Name is the name of the operating system. The supported values are linux and windows for now.
+	// Additional value may be defined in future and can be and one of:
+	// https://github.com/opencontainers/runtime-spec/blob/master/config.md#platform-specific-configuration
+	// Clients should expect to handle additional values and treat unrecognized values in this field as "",
```

Comment:
`treat unrecognized values in this field as ""` is a little weird, since "" should not be a valid value for this field. "Treat unrecognized values in this field as `os: null`" might be clearer and map more closely to how clients are already treating pods (since there's no `os` field today)?